presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -399,19 +401,31 @@ private synchronized QueryResults getNextResultWithRetry(long token, UriInfo uri
         boolean hasNotRetried = queryManager.getQueryRetryCount(queryId) < 1;
 
         if (historyBasedOptimizationEnabled && hasNotRetried && retryConditionsMet(queryResults) && retryQueryWithHistoryBasedOptimizationEnabled(session)) {
+            WarningCode warningFromOriginalQuery = new WarningCode(700, "Your query is being retried");
+
+            String originalQueryMessage = "This is your original query, it has failed and is being retried, which may change the query plan";
+            session.getWarningCollector().add(new PrestoWarning(warningFromOriginalQuery, originalQueryMessage));
+
             originalBeforeRetryQueryId = Optional.of(queryId);
             previousQueryTopLevelPlanHash = getCurrentTopLevelPlanHash();
             previousQueryFailureError = Optional.of(queryResults.getError());
         }
         else if (queryManager.getQueryRetryCount(queryId) == 1 && retryQueryWithHistoryBasedOptimizationEnabled(session) && retryConditionsMet(queryResults) && historyBasedOptimizationEnabled) {
             Optional<Integer> currentTopLevelPlanHash = getCurrentTopLevelPlanHash();
 
+            WarningCode warningFromRetryQuery = new WarningCode(701, "This is your retried query");
+
             if (previousQueryTopLevelPlanHash.isPresent() && currentTopLevelPlanHash.isPresent() && currentTopLevelPlanHash.equals(previousQueryTopLevelPlanHash)
                     || (!previousQueryTopLevelPlanHash.isPresent() && !currentTopLevelPlanHash.isPresent())) {
+                String retryQuerySamePlanMessage = "Since the plan hashes did not change, your retry query will not execute";
+                session.getWarningCollector().add(new PrestoWarning(warningFromRetryQuery, retryQuerySamePlanMessage));
+
                 queryManager.failQuery(queryId, new PrestoException(GENERIC_INTERNAL_ERROR, "Since the plan hashes did not change, your retry query will not execute." +
                         "Your original error was " + previousQueryFailureError.get() + ". Original QueryId: " + originalBeforeRetryQueryId +
                         ". Retry QueryId: " + queryId));
             }
+            String retryQueryMessage = "This query is a retry of the failed query: " + originalBeforeRetryQueryId + ".";
```

Comment:
I don't think we put period at the end of our warnings - double check

Reply:
https://www.internalfb.com/intern/presto/query/?query_id=20240724_225325_00679_vwf49#dr_presto in this warning we do, not sure if you count this

Reply:
Think in the message we have them just not in the title